presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcMetadata.java

```diff
@@ -107,12 +112,48 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
                 handle.getCatalogName(),
                 handle.getSchemaName(),
                 handle.getTableName(),
+                handle.getColumns(),
                 newDomain,
                 handle.getLimit());
 
         return Optional.of(new ConstraintApplicationResult<>(handle, constraint.getSummary()));
     }
 
+    @Override
+    public Optional<ProjectionApplicationResult<ConnectorTableHandle>> applyProjection(
+            ConnectorSession session,
+            ConnectorTableHandle table,
+            List<ConnectorExpression> projections,
+            Map<String, ColumnHandle> assignments)
+    {
+        JdbcTableHandle handle = (JdbcTableHandle) table;
+
+        List<JdbcColumnHandle> newColumns = assignments.values().stream()
+                .map(JdbcColumnHandle.class::cast)
+                .collect(Collectors.toList());
```

Comment:
ImmutableList

Reply:
`io.prestosql.util.MoreLists#mappedCopy` (requires move to plugin toolkit)

Reply:
> io.prestosql.util.MoreLists#mappedCopy (requires move to plugin toolkit)

That's overkill for such a simple usage.